Letca_Anna/week_02/Demo/Program.cs

```diff
@@ -33,24 +43,103 @@ static async Task Main(string[] args)
                 from menuItemResult1 in RestaurantDomain.CreateAndAddMenuItem("carbonara", 25, menu)
                 from menuItemResult2 in RestaurantDomain.CreateAndAddMenuItem("conpesto", 20, menu)
                 let menuItem2=(menuItemResult2 as MenuItemAdded)?.MenuItem
-                from clientResult in RestaurantDomain.CreateClient("gucdg34u6trgfh")
-                let client = (clientResult as ClientCreated)?.Client
-                from addToCartResult in RestaurantDomain.AddItemToCart(menuItem2, client)
-                from orderResult in RestaurantDomain.CreateAndPlaceOrder(client, restaurant)
+                from clientResult in RestaurantDomain.CreateClient("gucdg34u6trgfh","Anca")
+                //from orderResult in RestaurantDomain.CreateAndPlaceOrder(client, restaurant)
                 select restaurantResult;
 
-            //remove item from cart(item, client.cart)
-            //create and place order(client, restaurant)
-
-            Console.WriteLine(expr);
-
             var interpreter = new LiveInterpreterAsync(serviceProvider);
 
             var result = await interpreter.Interpret(expr, Unit.Default);
 
             var finalResult = result.Match<bool>(OnRestaurantCreated, OnRestaurantNotCreated);
             Assert.True(finalResult);
 
+            Restaurant foundRestaurant=null;
+            Menu foundMenu = null;
+            Client foundClient = null;
+
+            do
+            {
+                PrintInputOptions();
+                input = Console.ReadLine(); 
+                switch (input)
+                {
+                    case "1": // login (suppose client is already created/registered)
+                        var getClientExpr =
+                                        from client in RestaurantDomain.GetClient("gucdg34u6trgfh")
+                                        select client;
+                        var interpretedClient = await interpreter.Interpret(getClientExpr, Unit.Default);
+                        foundClient = (interpretedClient as ClientFound).Client;
+                        var clientResponse = interpretedClient.Match<bool>(OnClientFound, OnClientNotFound);
+                        Assert.True(clientResponse);
+                        break;
+                    case "2": //get restauraant
+                        Console.WriteLine("Enter the restaurant's name...");
+                        string restaurantName = Console.ReadLine();
+                        var getRestaurantExpr =
+                                        from restaurant in RestaurantDomain.GetRestaurant(restaurantName)
+                                        select restaurant;
+                        var interpretedRestaurant = await interpreter.Interpret(getRestaurantExpr, Unit.Default);
+                        foundRestaurant = (interpretedRestaurant as RestaurantFound).Restaurant;
```

Comment:
Same as above. Let's consider that an expression like 
``` csharp
(stuff as Type) 
```
something we want to avoid.